### PR TITLE
update base streaming rule cursor rule

### PR DESCRIPTION
**Pull request by jacknagz**
### Background

Committing an update to the cursor rule for writing streaming rules.

### Changes

- Refactored streaming rule cursor rule with examples
- Streaming rule caching cursor rule

### Testing

Sample prompt: `Create a rule that makes a low-severity alert any time someone assume's a role into AWS account 111111111111 via SSO from the dev-jn log source. Put the rule in @/production_security and use @streaming-rules.mdc `

Generated rule:

```python
from panther_aws_helpers import aws_cloudtrail_success

def rule(event):
    role_arn = event.deep_get("requestParameters", "roleArn", default="")
    return (
        aws_cloudtrail_success(event)
        and event.get("eventName") == "AssumeRole"
        and event.get("recipientAccountId") == "111111111111"
        and "AWSReservedSSO" in role_arn
    )

def title(event):
    actor = event.deep_get("userIdentity", "arn", default="unknown actor")
    role = event.deep_get("requestParameters", "roleArn", default="unknown role")
    return f"SSO role assumed into 111111111111: {actor} -> {role}"

def runbook(event):
    actor = event.deep_get("userIdentity", "arn", default="unknown actor")
    role = event.deep_get("requestParameters", "roleArn", default="unknown role")
    return f"""
1. Review CloudTrail events for [{actor}] and [{role}] in the aws_cloudtrail table around the time of this alert.
2. Confirm if the SSO role assumption was expected and authorized.
3. Investigate any follow-up actions performed by the assumed SSO role in account 111111111111.
4. If unauthorized, disable the session and rotate credentials as needed.
"""

def alert_context(event):
    return {
        "actor": event.deep_get("userIdentity", "arn", default=""),
        "role": event.deep_get("requestParameters", "roleArn", default=""),
        "timestamp": event.get("eventTime", ""),
        "source_ip": event.get("sourceIPAddress", ""),
        "user_agent": event.get("userAgent", ""),
        "action": event.get("eventName", ""),
    } 
```

### CODE REVIEW

File: .cursor/rules/streaming-rules.mdc [hunk 1]
```diff
@@ -0,0 +1,320 @@
...
+* `alert_context(event: dict) -> dict`: Important fields present in the events that expedite understanding critical information about the activity.
+
+### Dynamic Functions
+Only use the following functions when you need dynamic return values based on the events and always declare these functions with conditional logic:
+* `runbook(event: dict) -> str`: The specific and ordered steps to triage and resolve the alert. This is read by Panther's AI triage agent.
+* `severity(event: dict) -> str`: Dynamically adjust the severity of the alert to one of INFO, LOW, MEDIUM, HIGH, CRITICAL, or DEFAULT.
+* `destinations(event: dict) -> List[str]`: Dynamically set the list of destinations to send the alert to.
+
+# Rule Metadata
+The YML file has the following schema:
```
Comment (arielkr256) [inline .cursor/rules/streaming-rules.mdc]: Would it be more effective to tell cursor to read the yaml schema definition?  https://github.com/panther-labs/panther-analysis/blob/develop/.vscode/schemas/rule.json

File: .cursor/rules/streaming-rules.mdc [hunk 2]
```diff
@@ -0,0 +1,320 @@
...
+## Common Anti-patterns
+- **Don't implement thresholding** in rule logic (Panther handles this)
+- **Don't make titles too unique** (causes alert fragmentation)
+- **Don't access fields without safe methods** (causes KeyError exceptions)
+- **Don't use complex nested logic** (hurts performance and readability)
+- **Don't forget to test edge cases** (null values, missing fields)
+
+# Testing Strategy
+
+When creating rule tests, aim for at least one positive (`ExpectedResult: true`) and negative (`ExpectedResult: false`) case. Best practice is to cover all major code paths.
```
Comment (arielkr256) [inline .cursor/rules/streaming-rules.mdc]: When possible, use actual logs queried from the data lake (with sensitive fields redacted).

File: .cursor/rules/streaming-rules.mdc [hunk 3]
```diff
@@ -0,0 +1,320 @@
...
+        "actor": event.deep_get("userIdentity", "arn", default=""),
+        "timestamp": event.get("eventTime", ""),
+        "parameters": event.deep_get("requestParameters", default={}),
+        "access_key_id": event.deep_get("responseElements", "accessKey", "accessKeyId", default=""),
+        "action": event.get("eventName", ""),
+    }
+    return context
+```
+
+### `aws_iam_user_key_created_demo.yml`
```
Comment (arielkr256) [inline .cursor/rules/streaming-rules.mdc]: .yml filename/filepath should match .py in line 133

File: .cursor/rules/streaming-rules.mdc [hunk 4]
```diff
@@ -0,0 +1,320 @@
...
+1. Set `CreateAlert` to false
+2. Set `Severity` to INFO
+3. Only include the rule() method in the `.py` file
+4. Ignore alert-related metadata like deduplication
+
+# Examples
+
+## IAM User Backdoor Key Rule
+
+### `rules/aws_cloudtrail_rules/aws_iam_user_backdoor.py`
```
Comment (arielkr256) [inline .cursor/rules/streaming-rules.mdc:110]: filename should match line 183

File: .cursor/rules/streaming-rules.mdc [hunk 5]
```diff
@@ -0,0 +1,320 @@
...
+
+## Login Signal
+
+### panther_login_signal.py
+```python
+def rule(event):
+    return event.get("actionName", "") == "SIGN_IN"
+```
+
+### aws_iam_user_key_created_demo.yml
```
Comment (arielkr256) [inline .cursor/rules/streaming-rules.mdc]: Filename should match line 281

File: .cursor/rules/streaming-rules.mdc (hunk 1 above)
Comment (jacknagz) [reply .cursor/rules/streaming-rules.mdc]: Oh nice, I'll try to @ mention that file instead!

**Review by arielkr256 — APPROVED** [review]
LGTM!
